tests/qir_test.py

```diff
@@ -278,26 +287,30 @@ def test_multiple_successive_conditionals(
 
         circuit = circuit_from_qir(multiple_conditionals_bc_file_path)
 
-        non_condition_circuit = Circuit(6, 6)
+        non_condition_circuit = Circuit(circuit.n_qubits, circuit.n_bits)
         for com in circuit.get_commands()[:10]:
             non_condition_circuit.add_gate(com.op, com.args)
 
-        exp_non_condition_circuit = Circuit(6, 6)
+        exp_non_condition_circuit = Circuit(circuit.n_qubits, circuit.n_bits)
         for com in multiple_conditionals_circuit.get_commands()[:10]:
             exp_non_condition_circuit.add_gate(com.op, com.args)
 
         assert non_condition_circuit == exp_non_condition_circuit
 
         condition_com_1 = circuit.get_commands()[10]
+
+        inverse_reg_map = {v: k for k, v in circuit._reg_map.items()}
+        assert inverse_reg_map[condition_com_1.args[0]].reg_name == "%0"
+
         condition_circuit_1 = condition_com_1.op.op.get_circuit()
         condition_circuit_1_coms = condition_circuit_1.get_commands()
 
         exp_condition_com_1 = multiple_conditionals_circuit.get_commands()[10]
         exp_condition_circuit_1 = exp_condition_com_1.op.op.get_circuit()
         exp_condition_circuit_1_coms = exp_condition_circuit_1.get_commands()
 
-        non_condition_circuit_1 = Circuit(6, 6)
-        exp_non_condition_circuit_1 = Circuit(6, 6)
+        non_condition_circuit_1 = Circuit(6, 262)
```

Comment:
Is this blow-up in circuit size (from 6 to 262) necessary?

Reply:
Yes, otherwise I get the error that the circuit doesn't hold enough unitids.

```sh
        for com in condition_circuit_1_coms:
            if not isinstance(com.op, Conditional):
>               non_condition_circuit_1.add_gate(com.op, com.args)
E               RuntimeError: Circuit does not contain unit with id: c[257]
```